custom_model_runner/datarobot_drum/drum/memory_monitor.py

```diff
@@ -22,10 +22,33 @@ def __init__(self, pid=None, include_childs=True):
         self._include_childs = include_childs
 
     def collect_memory_info(self):
-        current_process = psutil.Process(os.getpid())
-        proc_rss_mb = ByteConv.from_bytes(current_process.memory_info().rss).mbytes
-        for child in current_process.children(recursive=True):
-            proc_rss_mb += ByteConv.from_bytes(child.memory_info().rss).mbytes
+        def get_proc_data(p):
+            return {
+                "pid": p.pid,
+                "cmdline": p.cmdline(),
+                "mem": ByteConv.from_bytes(p.memory_info().rss).mbytes,
+            }
+
+        drum_process = None
+        nginx_rss_mb = 0
+        drum_rss_mb = 0
+
+        for proc in psutil.process_iter():
+            if "drum" in proc.name().lower():
+                if "server" in proc.cmdline():
+                    drum_process = proc
+
+        if drum_process:
+            drum_rss_mb = ByteConv.from_bytes(drum_process.memory_info().rss).mbytes
+            drum_info = []
+            drum_info.append(get_proc_data(drum_process))
+            for child in drum_process.children(recursive=True):
+                drum_rss_mb += ByteConv.from_bytes(child.memory_info().rss).mbytes
+                drum_info.append(get_proc_data(child))
+
+        for proc in psutil.process_iter():
+            if "nginx" in proc.name().lower():
```

Comment:
I feel like this is also very brittle. I have a lot of nginx processes running locally normally.

Reply:
Generally it is not for you, It's for checking memory consumption in an env.
And, as you note, it may be brittle. Better way to see memory consumption is in pod stats, but we also can run locally or locally in docker. So there is no single solution.

Reply:
Let me know if you want to brainstorm but I don't think this is a valid solution.